### PR TITLE
190916 default tasks

### PR DESCRIPTION
**Pull request by zrolfs**
Create functionality that allows a user to save their own set of custom defaults for routine analyses. #1349

### CODE REVIEW

File: GUI/CalibrateTaskWindow.xaml [hunk 1]
```diff
@@ -376,6 +376,12 @@
         <StackPanel  Orientation="Horizontal" HorizontalAlignment="Center" Grid.Row="1" Margin="5">
             <Button x:Name="cancelButton" Content="Cancel" Click="CancelButton_Click" />
             <Button x:Name="saveButton" Content="Save Calibrate Task" Click="SaveButton_Click" FontSize="24" />
+            <Button x:Name="setDefaultbutton" Click="SaveAsDefault_Click" Width="100" >
+                <TextBlock Text="Save As Default" TextWrapping="Wrap" TextAlignment="Center"/>
+                <ToolTipService.ToolTip>
+                    <ToolTip Content="Save these parameters as the default. MetaMorpheus can return to original defaults by deleting the folder 'DefaultParameters'" ToolTipService.ShowDuration="999999" ToolTipService.InitialShowDelay="500" />
```
Comment (trishorts) [inline GUI/CalibrateTaskWindow.xaml]: rather have a button reset (which deletes the folder) than have the user have to track it down. i sometimes can't find these folders. same comment on similar items below

**Comment by rmillikin** [comment]
how does this work when MM gets updated and new variables are added to, say, Common Params? .toml files aren't guaranteed to be backwards/forwards compatible

**Comment by zrolfs** [comment]
> how does this work when MM gets updated and new variables are added to, say, Common Params? .toml files aren't guaranteed to be backwards/forwards compatible

There are a few different paths; I'll outline each one and describe how they're handled:
1. The user never sets a toml as default. If a toml doesn't exist in the default folder when a new task is created, then a toml is NOT created in that folder (i.e. the folder remains empty). The new task simply calls the method that creates a new task. This is the same functionality as we currently have in MM.
2. The user has previously set a toml as the default but it is no longer supported. A GUI warning message is displayed telling the user that it was unable to read the old toml and the MM original default settings are used.

**Comment by rmillikin** [comment]
it seems that `Toml.ReadFile` is used to read the .toml and if there is an exception in that method, a warning is printed. 

1. does `Toml.ReadFile` throw an exception if there is a key/value pair in the .toml that does not correspond to a variable in the object it's creating?
2. if there is a new variable in CommonParams, does `Toml.ReadFile` throw an exception if there is no value in the .toml for that variable?

**Comment by zrolfs** [comment]
1. Yes
2. No; the default value for the new variable is used.

**Comment by rmillikin** [comment]
I guess that's acceptable behavior for now. It does seem like a bad idea to use the CommonParam default for any new settings without printing out a warning, but I think that's what we do when someone drags in an old .toml. I guess my issue with this PR is that dragging in an old .toml is deliberate, while using a default .toml can quickly get out of date without the user really thinking about potential problems there. 

Ultimately, it would be a good idea to create a method that, upon loading a .toml file, warnings are added to the notification area if variables are not in the .toml.

**Comment by zrolfs** [comment]
Any variable that has a default of null is not written to a toml, so we would have lots of false-positive warnings.

**Pull request merged**